misc/python/materialize/data_ingest/executor.py

```diff
@@ -0,0 +1,359 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import json
+from typing import Any, Dict, List
+
+import pg8000
+from confluent_kafka import Producer  # type: ignore
+from confluent_kafka.admin import AdminClient, NewTopic  # type: ignore
+from confluent_kafka.schema_registry import Schema, SchemaRegistryClient  # type: ignore
+from confluent_kafka.schema_registry.avro import AvroSerializer  # type: ignore
+from confluent_kafka.serialization import (  # type: ignore
+    MessageField,
+    SerializationContext,
+)
+
+from materialize.data_ingest.data_type import Backend
+from materialize.data_ingest.field import Field
+from materialize.data_ingest.row import Operation
+from materialize.data_ingest.transaction import Transaction
+
+
+class Executor:
+    def run(self, transactions: List[Transaction]) -> None:
+        raise NotImplementedError
+
+    def print_progress(self, cur_transaction: int, num_transactions: int) -> None:
+        if (100 * cur_transaction) % num_transactions == 0:
+            print(
+                f"{type(self).__name__}: {int((100 * cur_transaction) / num_transactions):2d}% ({cur_transaction} / {num_transactions})"
+            )
+
+
+class PrintExecutor(Executor):
+    def run(self, transactions: List[Transaction]) -> None:
+        for transaction in transactions:
+            print("Transaction:")
+            print("  ", transaction.row_lists)
+
+
+def idfn(d: str, ctx: Any) -> str:
```

Comment:
Is this a common (Python) name I am not aware of? If not, can we find a more expressive name?

Reply:
That's obviously the identity function :D Sorry for the Haskell-style code